streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -187,9 +189,13 @@ class CheckpointWriter(
     val bos = new ByteArrayOutputStream()
     val zos = compressionCodec.compressedOutputStream(bos)
     val oos = new ObjectOutputStream(zos)
-    oos.writeObject(checkpoint)
-    oos.close()
-    bos.close()
+    try {
+      oos.writeObject(checkpoint)
+    } finally {
+      oos.close()
+      bos.close()
```

Comment:
Here's it's sufficient to close `oos` only. `bos.close()` is a no-op anyway, but chained writers close their delegates.